app/controllers/catalog_controller.rb

```diff
@@ -0,0 +1,40 @@
+require 'moab/stanford'
+
+##
+# CatalogController allows clients to manipulate the Object Inventory Catalog, e.g.
+# to add an existing moab object to the catalog, or to update an entry for a moab object
+# that's already in the catalog.
+class CatalogController < ApplicationController
+  before_action :set_id
+
+  def add_preserved_object
+    PreservedObject.create!(
+      druid: @id, size: moab_size(@id), current_version: Stanford::StorageServices.current_version(@id)
+    )
+  end
+
+  def update_preserved_object
+    preserved_obj = PreservedObject.find_by(druid: @id)
+    preserved_obj.update_attributes(
+      druid: @id, size: moab_size(@id), current_version: Stanford::StorageServices.current_version(@id)
+    )
+  end
+
+  private
+
+  def set_id
+    @id = catalog_params[:id]
+    head(:unprocessable_entity) if @id.blank?
```

Comment:
So this would respond with a 422 Unprocessable Entity. Why did you decide to use this instead of a regular 400?

https://tools.ietf.org/html/rfc7231#section-6.5.1

Reply:
this was something that we weren't totally sure about, but it seemed like the most recommended error code from the few stack exchange posts and such that we found, e.g. https://stackoverflow.com/a/10323055:
> The 422 (Unprocessable Entity) status code means the server understands the content type of the request entity (hence a 415 (Unsupported Media Type) status code is inappropriate), and the syntax of the request entity is correct (thus a 400 (Bad Request) status code is inappropriate) but was unable to process the contained instructions. For example, this error condition may occur if an XML request body contains well-formed (i.e., syntactically correct), but semantically erroneous, XML instructions.

from https://tools.ietf.org/html/rfc4918#page-78

though the RFC you link is newer (2014 vs 2007).  and, now that i scroll up, i notice that maybe the spec we made that decision on is WebDAV specific.

400 felt a bit generic to me, and not quite right based on the short description in the RFC, but it did feel like the next most correct thing besides 422, so if you still feel that 422 is wrong, i think we're happy to switch over to 400.

Reply:
OK, sounds like this is really a toss-up.